### DISCUSSION TITLE
What is Planetscale doing to protect against data loss?

### DISCUSSION THREAD

**Discussion by brettscott** [planetscale/discussion#321] [Documentation] [answered]
I haven't seen documentation or discussions on data loss and how PlanetScale/Vitess protects against it.   

Service/data availability is defined in the Service Level Agreement (SLA), but not data loss.

Daily scheduled backups and manually-invoked backups ensure no more than 1 days data is lost.  I'm interested in understanding the risk of data loss in the < 24 hours window.

1. What is the level of certainty that an INSERT query with corresponding acknowledgement to the client can be considered saved to disk and "safe"?

2.  What's the architecture or mechanism that ensures there is no PlanetScale MySQL database data loss if one of the following has failed: a) one hard disk, b) one compute or c) one data centre?

3.  Do all tiers/plans receive the same level of protection against data loss?

**Comment by fooforge** [comment, marked as answer]
Hi @brettscott,

My apologies for the delay in answering! This thread has slipped through the cracks which shouldn't have happened, but it did.

There's quite a list of things we do to ensure your data is safe with us:

1. **Automatic backups**
  Every database on PlanetScale gets backed up every 24 hours by default. On the Scaler plan (and up) we automatically back up your data every 12 hours.
2. **Manual backup (schedules)**
  You can create additional [manual backups](https://planetscale.com/docs/concepts/back-up-and-restore#create-manual-backups) with control over their retention period, and create entire custom backup schedules. Our [CLI tool `pscale`](https://github.com/planetscale/cli) allows you to dump your data periodically as well through `pscale database dump [database] [branch]` from a remote location of your choice.
3. **Availability zones**
  Every database hosted on PlanetScale is a Vitess cluster. For [production branches](https://planetscale.com/docs/concepts/branching#development-and-production-branches) we spin up instances in two different availability zones. These zones are isolated locations within a region.
4. **Read-only regions**
  You also have the ability to add up to two [read-only regions](https://planetscale.com/docs/concepts/read-only-regions) outside of your core region for your database. While the feature was built to support globally distributed applications better by enabling your database to perform low latency reads in the regions closest to your applications and users, it obviously also acts as yet another safety net preventing catastrophic data loss.

We're SOC2 Type II compliant, our employees' access to critical systems is tightly audited and time-gated, and your data is AES encrypted both in transit and at rest. You can find further information on security and compliance [in our documentation](https://planetscale.com/docs/concepts/security).

There's likely more to say about how we build resilient infrastructure and I may have mixed up the topics of data resilience and high availability a tad too much, but I hope you find my response helpful nonetheless. For a more in-depth conversation, feel free to reach out to our Sales team via https://planetscale.com/contact.

**Reply by brettscott** [reply]
Thanks for the details response @fooforge.   This has indeed answered by questions, particular the detail on AZs and read-only zones.  Much appreciated.
